aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -27,7 +30,10 @@ Tensor& add_out(Tensor& result, const Tensor& self, const Tensor& other, Scalar
   }
   auto iter = TensorIterator::binary_op(result, self, other,
     /*check_internal_overlap=*/true);
+  TORCH_CHECK(isFloatingType(iter.dtype()) || alpha.isIntegral(), alpha_mismatch_err);
   add_stub(iter.device_type(), iter, alpha);
+  AT_ASSERT(result.scalar_type() == iter.output().dtype());
+  result.copy_(iter.output());
```

Comment:
Why do you need this .copy_ in in-place ops? It's not obvious to me from the code

Reply:
The `iter->output()` is owned by TensorIterator, so can't just be assigned to the result reference (when I tried that I got segfaults, I believe caused by destruction of the `TensorIterator`, `OperandInfo`s, and `Tensor`). Is there some other way to solve this? Would you prefer I add some comments?  The only alternative I could think of was if the `OperandInfo` held a pointer to a `Tensor` and was a non-owner, but I'm not sure that would be clearer. I believe this worked before the type promotion changes because we weren't calling `to()` and so the provided result tensor was what got returned by the `iter->output`.

Reply:
This needs to be fixed before the PR can be merged. I'm concerned about two things: 1) If someone forgets this, their code will break silently with mixed-type operations and 2) the overhead of the copy invocation even when the result and iter.output() are the same

I can suggest alternatives if you'd like help

Reply:
removed the copy_ calls